src/components/common/InAppBrowserWarningModal.tsx

```diff
@@ -50,16 +50,21 @@ const InAppBrowserWarningModal = () => {
   if (isLineInAppBrowser) window.location.assign(lineOpenExternalBrowserUrl)
 
   const isRemind = sessionStorage.getItem('kolable.InAppBrowserWarning.isRemind')
-  if (isInAppBrowser && !isToast && !isRemind) {
-    toast({
-      title: formatMessage(commonMessages.InAppBrowserWarningModal.warning),
-      status: 'warning',
-      isClosable: true,
-      position: 'bottom',
-    })
-    setIsToast(true)
-    sessionStorage.setItem('kolable.InAppBrowserWarning.isRemind', 'true')
-  }
+  const toastId = 'inAppBrowser'
+
+  useEffect(() => {
+    if (isInAppBrowser && !toast.isActive(toastId) && toastRef.current == null && !isRemind) {
+      sessionStorage.setItem('kolable.InAppBrowserWarning.isRemind', '1')
+
+      toastRef.current = toast({
+        id: toastId,
+        title: formatMessage(commonMessages.InAppBrowserWarningModal.warning),
+        status: 'warning',
+        isClosable: true,
+        position: 'bottom',
+      })
+    }
+  }, [isRemind])
```

Comment:
If using the toast directly, it might impact React rendering, so it's advisable to utilize useEffect to prevent side effects. Consequently, if you only want the component to render once, isRemind is a suitable option.

Reply:
What I mean is that you should fill in the dependency values ​​correctly according to ESLint's instructions. Don't fill in the dependency values ​​based on human judgment. It will be safer.

Reply:
You can read this [article](https://ithelp.ithome.com.tw/articles/10305701) and this [article](https://ithelp.ithome.com.tw/articles/10306185)